### DISCUSSION TITLE
Can't get autobuild to work :(

### DISCUSSION THREAD

**Discussion by Alraune22** [James-Yu/LaTeX-Workshop#4037] [Q&A]
Hi, I'm a biochemistry students that decided to try his luck with LaTeX workshop. 
I like the extension a lot so far, but as my title said, I can't get the autobuild to work properly. 

Form what I understand from the documentation, I am supposed to add "latex-workshop.latex.autoBuild.run":"onFileChange" to my settings.json file. I also wanted to add an intervall of 5 seconds before it does it, so I tried to add that too. The "workbench.colorTheme": "Default Dark Modern", was already present, I just added the coma at the end. 

{
    "workbench.colorTheme": "Default Dark Modern",
    "latex-workshop.latex.recipe.default":"first",
    "latex-workshop.latex.autoBuild.run":"onFileChange",
    "latex-workshop.latex.autoBuild.interval":5000,
}

![image](https://github.com/James-Yu/LaTeX-Workshop/assets/134406039/c23456ff-532e-4fdc-971e-647a98774f18)

I would love to get the document autocompiled, it would add a lot of convenience to my LaTeX journey 
Greetings from switzerland

**Comment by James-Yu** [comment]
Please provide more context, or you can file an issue in this repository. Also, `latex-workshop.latex.autoBuild.interval` is an interval between two builds, instead of a delay after file change.

**Reply by Alraune22** [reply]
thanks for the quick reply
My problem is, that even if I cange anything in the .tex file, it will not autobuild the .pdf file. 

{
    "workbench.colorTheme": "Default Dark Modern",
    "latex-workshop.latex.autoBuild.run":"onFileChange",
}

I changed the settings.json file because I missunderstood the interval recipe, this is the new one. 
![image](https://github.com/James-Yu/LaTeX-Workshop/assets/134406039/d9eb2369-41d1-41dd-8879-89cf93b72b9e)
As far as I understand, if I now change a line in my .tex file, it should automatically generate a new .pdf

![image](https://github.com/James-Yu/LaTeX-Workshop/assets/134406039/8eac18be-3495-48fe-bde1-d6dde4e94326)
So if I add a new item to the list, LaTeX Workshop should automatically generate the new pdf with the new item in the list. 
Sadly, it does not do that and I do not understand why it doesn't. 

Thank you for your patience, I have never before used VS code and all the tutorials concerning stuff like this are not really helpful :)
Greetings

**Reply by James-Yu** [reply]
The save is triggered on changes to the file. Changes to the editor won’t work in your case.

**Reply by Alraune22** [reply]
AAAAAAAH, i get it, thank you very much!
I activated the autosave function of VS code, this triggers the build automatically after I changed something in the editor, because it saves the file. I wonder if there is a "correct" way to do this or if this the intended way...
Thanks for your patience!

**Reply by James-Yu** [reply]
Indeed, there is no correct or wrong way of doing this. Yours is absolutely fine if that fits your need. I myself manually save tex files, but auto-save may also be fine as long as it does not trigger too many builds.

**Reply by Alraune22** [reply]
Alright, thank you so much, have a nice day <3
